org.idempiere.test/src/org/idempiere/test/base/ReportTest.java

```diff
@@ -0,0 +1,77 @@
+ * http://www.idempiere.org                                            *
+ *                                                                     *
+ * Copyright (C) Contributors                                          *
+ *                                                                     *
+ * This program is free software; you can redistribute it and/or       *
+ * modify it under the terms of the GNU General Public License         *
+ * as published by the Free Software Foundation; either version 2      *
+ * of the License, or (at your option) any later version.              *
+ *                                                                     *
+ * This program is distributed in the hope that it will be useful,     *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of      *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the        *
+ * GNU General Public License for more details.                        *
+ *                                                                     *
+ * You should have received a copy of the GNU General Public License   *
+ * along with this program; if not, write to the Free Software         *
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,          *
+ * MA 02110-1301, USA.                                                 *
+ *                                                                     *
+ * Contributors:                                                       *
+ * - Carlos Ruiz - globalqss	       								   *
+ **********************************************************************/
+package org.idempiere.test.base;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.File;
+
+import org.compiere.model.MOrder;
+import org.compiere.model.MPInstance;
+import org.compiere.model.MProcess;
+import org.compiere.process.ProcessInfo;
+import org.compiere.process.ServerProcessCtl;
+import org.compiere.util.Env;
+import org.idempiere.test.AbstractTestCase;
+import org.junit.jupiter.api.Test;
+
+/**
+ * @author Diego Ruiz - BX Service GmbH
+ */
+public class ReportTest extends AbstractTestCase {
+
+	public ReportTest() {
+	}
+
+	private static final int Order_Print_Process = 110;
+
+	/**
+	 * https://idempiere.atlassian.net/browse/IDEMPIERE-6165
+	 */
+	@Test
+	public void testPDFFileName() {
+		MProcess orderReport = MProcess.get(Env.getCtx(), Order_Print_Process);
+		MOrder order = new MOrder(Env.getCtx(),  108, getTrxName()); // Garden Order 60000
+		
+		String tmpdirname = System.getProperty("java.io.tmpdir") + System.getProperty("file.separator");
+		String fileName = order.getDocumentNo() + ".pdf";
+
+		ProcessInfo pi = new ProcessInfo(orderReport.getName(), orderReport.getAD_Process_ID());
+		pi.setRecord_ID(order.getC_Order_ID());
+		pi.setAD_Client_ID(Env.getAD_Client_ID(Env.getCtx()));
+		pi.setTable_ID(order.get_Table_ID());
+		pi.setPrintPreview(true);
+		pi.setIsBatch(true);
+		pi.setPDFFileName(tmpdirname + fileName);
```

Comment:
thinking further, this is thread safe. In other words, if a process that did this is run concurrently by different user, the output file will conflict with each other.

Reply:
Just added a new commit - following the same approach as the core. A temp folder with the date and session id creating it, so it is not duplicated. 

By the way, I believe this code taken from FileUtils

```
Calendar cal = Calendar.getInstance();
		SimpleDateFormat sdf = new SimpleDateFormat("yyyyMMddHHmmssSSS");
		String dt = sdf.format(cal.getTime());
		String tmpdirname = System.getProperty("java.io.tmpdir");
		tmpdirname += System.getProperty("file.separator") + "rpttmp_" + dt + "_" + Env.getContext(Env.getCtx(), Env.AD_SESSION_ID) + System.getProperty("file.separator");

```

It should be moved to a method (it is repeated and FileUtils), so it can be reused everywhere, including the Unit Tests. I didn't want to do that in this ticket to not mix things up. Should I create a new ticket for that?

Reply:
I would think it is less prone to developer error that setPDFFileName(fileName) just take the pdf file name without path and the create unique temp folder auto happens in ReportEngine instead.

If needed, the re-factoring of the FileUtil code can happens here.

Reply:
to avoid breaking old plugins that write the file in some folder - I think the suggestion by @hengsin must be implemented just if the fileName doesn't contain slash or backslash ("/" or "\") the known file separators.

Reply:
hmm ... isn't this a new added behaviour ?

Reply:
it's something new for core, but it was working correctly for jasper

Reply:
@CarlosRuiz-globalqss / @hengsin 
I just updated the pull request with all the suggestions made on this thread. 

Please let me know if something else needs to be changed